packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/listFeatures.ts

```diff
@@ -156,21 +155,22 @@ function isAListPattern(textBeforeCursor: string) {
  * AutoBullet edit feature, provides the ability to automatically convert current line into a list.
  * When user input "1. ", convert into a numbering list
  * When user input "- " or "* ", convert into a bullet list
+ * @deprecated
  */
 const AutoBullet: BuildInEditFeature<PluginKeyboardEvent> = {
     keys: [Keys.SPACE],
     shouldHandleEvent: (event, editor) => {
-        if (!cacheGetListElement(event, editor)) {
-            const searcher = editor.getContentSearcherOfCursor(event);
-            const textBeforeCursor = searcher.getSubStringBefore(5);
-            const listTrigger = (text: string) =>
-                editor.isFeatureEnabled(ExperimentalFeatures.AutoFormatList)
-                    ? getListInfo(text)
-                    : isAListPattern(text);
+        if (
+            !cacheGetListElement(event, editor) &&
+            !editor.isFeatureEnabled(ExperimentalFeatures.AutoFormatList)
+        ) {
+            let searcher = editor.getContentSearcherOfCursor(event);
+            let textBeforeCursor = searcher.getSubStringBefore(4);
+
             // Auto list is triggered if:
             // 1. Text before cursor exactly matches '*', '-' or '1.'
             // 2. There's no non-text inline entities before cursor
-            return !searcher.getNearestNonTextInlineElement() && listTrigger(textBeforeCursor);
+            return isAListPattern(textBeforeCursor) && !searcher.getNearestNonTextInlineElement();
         }
```

Comment:
can we not change the code of this one since we are about to deprecate it, as long as it still work. So that people who is still using this code won't be impacted.